src/Server/src/Core/Encryption/KeyvaultEncryption/AzureKeyVaultExtension.cs

```diff
@@ -0,0 +1,27 @@
+using Azure.Security.KeyVault.Keys.Cryptography;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+
+namespace IdOps;
+
+public static class AzureKeyVaultExtension
+{
+    public static IServiceCollection AddAzureKeyVault(
+        this IServiceCollection services,
+        IConfiguration configuration)
+    {
+        AzureKeyVaultOptions options = configuration
+            .GetSection("IdOps:Azure:SecretEncryption:AzureKeyVault")
+            .Get<AzureKeyVaultOptions>();
+
+        services.AddSingleton(options);
+        services.AddSingleton<ICryptographyClientProvider, AzureKeyvaultCryptographyClientProvider>();
+        services.TryAddSingleton<CryptographyClient>(provider =>
+        {
+            var cryptographyClientProvider = provider.GetService<ICryptographyClientProvider>();
+            return cryptographyClientProvider.GetCryptographyClientAsync().GetAwaiter().GetResult();
```

Comment:
Lets avoid using `GetAwaiter().GetResult();` by not registering the CryptographyClient via dependency injection but instead using the ICryptographyClientProvider directly in `KeyVaultEncryptionProvider`. The implementation of ICryptographyClientProvider can use internal caching to avoid having to asynchronously recreate the CryptographyClient multiple times. If we want to eager load the provider we could start the task of loading the CryptographyClient by creating a new task in the ctor of CryptographyClient.